.github/workflows/build.yml

```diff
@@ -0,0 +1,99 @@
+name: Build
+on: ['push','pull_request']
+
+jobs:
+  install:
+    name: Install
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/cache@v1
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-node_modules-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-node_modules-
+      - name: Use Node.js 11.12
+        uses: actions/setup-node@v1
+        with:
+          node-version: 11.12
+      - name: Install
+        run: yarn
+  test:
+    needs: install
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Use Node.js 11.12
+        uses: actions/setup-node@v1
+        with:
+          node-version: 11.12
+      - uses: actions/cache@v1
+        id: cypress_cache
+        with:
+          path: ~/.cache
+          key: cypress_cache-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-cypress_cache-
+      - uses: actions/cache@v1
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-node_modules-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-node_modules-
+      - name: Install
+        run: npx cypress install
+      - name: Lint
+        run: |
+          yarn prettier:check
+          yarn prettier:json:check
+          yarn lint
+      - name: install libconf
+        run: sudo apt-get install libgconf-2-4
+      - name: Test
+        run: |
+          yarn test:cypress:ci
+  build:
+    needs: install
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Use Node.js 11.12
+        uses: actions/setup-node@v1
+        with:
+          node-version: 11.12
+      - uses: actions/cache@v1
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-node_modules-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-node_modules-
+      - name: Build
+        run: |
+          yarn build
+          echo VUE_APP_TRAVIS_COMMIT=$GITHUB_SHA >> .env.production
+      - name: Upload Build
+        uses: actions/upload-artifact@master
+        if: github.ref == 'refs/heads/master'
+        with:
+          name: build
+          path: dist
+  deploy:
+    needs: ['build','test']
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master'
+    steps:
+      - name: Download Build
+        uses: actions/download-artifact@master
+        if: github.ref == 'refs/heads/master'
+        with:
+          name: build
+          path: dist
+      - name: GitHub Pages Deploy
+        uses: zekth/gh-deploy-action@1.0.0
+        with:
+          dir: dist
+          gh_token: ${{ secrets.GH_TOKEN }}
```

Comment:
I don't have permissions to create one of these. Where do we get these from?

Reply:
In this case its provided for you <https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token>

Reply:
You can create one for the qmk_bot in the security settings of it

Reply:
Or like zvecr said you can also use the generated one. I think it will work also

Reply:
Yeah I'm using it with a different action and it seems fine on push master

Reply:
I'll change it. Thanks @zvecr